library/spit_results.py

```diff
@@ -251,17 +251,28 @@ def write_to_csv(self):
         # Make sure the controller expanded the default option.
         assert self.fact_names != ['default']
 
-        keys = set(self.fact_names)
-
-        # Special processing for JBoss facts.
-        for _, host_vars in iteritems(self.all_vars):
-            uuid = host_vars['connection.uuid']
-            host_vals = safe_next((vals
-                                   for vals in self.vals
-                                   if vals['connection.uuid'] == uuid))
-
-            host_vals.update(process_jboss_versions(host_vars))
-            host_vals.update(process_addon_versions(host_vars))
+        fact_keys = ['connection.host', 'connection.port', 'connection.uuid']
+        if isinstance(self.fact_names, list):
```

Comment:
should always be, moved this code over from run_cmds.py so I could unblock my merge conflicts. This section of code will go away when I remove the fact group prefixing (i.e. Username_uname.os; will get rid of Username_). Then self.facts_names should be what we want.